.github/workflows/ci.yml

```
@@ -1,180 +1,225 @@
+## The main Github Actions workflow
 name: CI
 
-## Only run when:
-##   - manually triggered
-##   - PR's are (re)opened
-##   - push to master (i.e. merge develop -> master)
-
 on:
   push:
     branches:
       - master
-  pull_request:
+      - develop
+      - next
+    paths-ignore:
+      - "**.md"
+      - "**.yml"
   workflow_dispatch:
     inputs:
       tag:
         description: "The tag to create (optional)"
         required: false
+  pull_request_target:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    paths-ignore:
+      - "**.md"
+      - "**.yml"
+    ## might be better to use inclusive v exclusive paths here, ex:
+    # paths:
+    #   - "**.rs"
+    #   - "**.clar"
+  pull_request_review:
+    types:
+      - submitted
+
+defaults:
+  run:
+    shell: bash
 
 concurrency:
-  group: ${{ github.head_ref || github.run_id }}
+  group: ci-${{ github.head_ref || github.ref }}
```

Is this intended to be:
```suggestion
  group: ci-${{ github.head_ref || github.ref || github.run_id}}
```
 following the concurrency pattern set in your other workflows?